src/Provider/DefaultSmsProvider.php

```diff
@@ -216,6 +203,8 @@ public function processDeliveryReport(Request $request, SmsGatewayInterface $sms
    * @return \Drupal\sms\Entity\SmsGatewayInterface|null
    *   A SmsGateway config entity, or NULL if default gateway is not set or
    *   invalid.
+   *
+   * @deprecated remove. shouldnt rely on this since its really a fallback.
```

Comment:
Like I think some parts of the framework rely on this API too much, so if there is an invalid gateway then the module may crash the site..